README.md

```diff
@@ -1 +1,109 @@
+
+Like before, we order a hot chocolate. The cashier takes your order, updating the Model. They then shout to the barista that they need a hot chocolate. The barista makes the hot chocolate, and returns it to the cashier. The cashier then writes your name on the lid, and returns it to you.
+
+The differences here is that we no longer have one person taking your order and delivering your drink. Instead, the cashier (ViewModel) uses a helper (business logic) to get your drink order (Model) completed and only modifies the lid with your name (view logic) and returns it to you (the View).
+
+This is the preferred architecture for Angular applications. Why? We no longer have a controller doing everything for us. Instead, we can simplify the application down massively by having helpers that we can use everywhere.
+
+## Data-binding
+
+Data-binding is the term for automatic synchronisation of data between the model and the view. This comes in two formats - one-way and two-way data-binding.
+
+### One-way data-binding
+
+Imagine you have a form input, asking for the user's name. You've also got a `<h2>` tag below, that you want to populate to say "Hey, {name}!". Wouldn't it be neat if this `<h2>` tag could update as the user types in their name? This is called one-way data-binding. The model (our user's name) is updated and our view is updated to reflect the change to our model.
+
+For example, you'd use one-way data-binding when you've got a search box - the user can type in their search query and the title can change to "Results for: {searchQuery}" - pretty cool!
+
+[Here's an example of one-way data-binding.](https://jsfiddle.net/6z6d5b7x/) 
+
+### Two-way data-binding 
+
+Two-way data-binding occurs when you bind your model value to an element that can both change and display the value of the model (such as an input).
+
+You aren't restricted to updating your model value just in the `<input />` however - you can update this Model value elsewhere (such as a "reset" button to reset the model) and it will be automatically reflected in the input (it will go blank).
+
+[Checkout this example of two-way data-binding](https://jsfiddle.net/6z6d5b7x/1/)
+
+You can see the flow of this here -
+
+![Two-way data-binding](https://docs.angularjs.org/img/Two_Way_Data_Binding.png)
+
+Two-way data-binding has a one-directional flow of data which ensures that the View and the Model data is always synchronized.
+
+In the above example our template compiles to become our View, which is a representation of the Model data. The View can then change our Model (for instance, an `<input />` being updated), which then updates our View. Model changes drive View changes and View changes drive Model changes. This keeps everything always synchronized. 
+
+## Dependency Injection
```

Comment:
This feels like it should be in a different README later on - we don't need to know about Dependency Injection yet, let this one just be about MVVM and data binding, this feels like too much of a left turn